Cython/Compiler/FusedNode.py

```diff
@@ -929,8 +929,11 @@ def generate_function_definitions(self, env, code):
             self.py_func.pymethdef_required = True
             self.fused_func_assignment.generate_function_definitions(env, code)
 
+        from . import Options
         for stat in self.stats:
-            if isinstance(stat, FuncDefNode) and stat.entry.used:
+            if (isinstance(stat, FuncDefNode)
+                and (stat.entry.used
+                     or (Options.cimport_from_pyx and not stat.entry.visibility == 'extern'))):
```

Comment:
IMHO those multi-factor conditions are not really all that readable and hiding some factors into a variable makes it better
```suggestion
            from_pyx = Options.cimport_from_pyx and not stat.entry.visibility == 'extern'
            if isinstance(stat, FuncDefNode) and (stat.entry.used or from_pyx):
```